consul/kv_consul.go

```diff
@@ -141,6 +126,40 @@ func New(
 	}, nil
 }
 
+// New constructs a new kvdb.Kvdb given a list of end points to conntect to.
+func New(
+	domain string,
+	machines []string,
+	options map[string]string,
+	fatalErrorCb kvdb.FatalErrorCB,
+) (kvdb.Kvdb, error) {
+	var kv kvdb.Kvdb
+	var err error
+	if len(machines) == 0 {
```

Comment:
nit: the loop is repeated without the checks for http/https.
It would be simpler to just reset machines to defaultMachines, its a copy to pointer.

Reply:
also, add an incorrect ip in default machines to test this out ?

Reply:
done and done